src/capybara.js

```diff
@@ -440,9 +473,34 @@ Capybara = {
     mouseTrigger('mousemove', options);
     mouseTrigger('mouseup', options);
   },
-
+  registerNode: function(node) {
+    this.nextIndex++;
+    this.nodes[this.nextIndex] = node;
+    return this.nextIndex;
+  },
   equals: function(index, targetIndex) {
     return this.getNode(index) === this.getNode(targetIndex);
+  },
+  _visitedObjects: [],
+  wrapResult: function(arg) {
+    if (this._visitedObjects.indexOf(arg) >= 0) { return '(cyclic structure)'; }
+    if (arg instanceof NodeList) { arg = Array.prototype.slice.call(arg, 0); }
+    if (Array.isArray(arg)) {
+      for(var _j = 0; _j < arg.length; _j++) {
+        arg[_j] = this.wrapResult(arg[_j]);
+      }
+    } else if (arg && arg.nodeType == 1 && arg.tagName) {
+      return {'element-581e-422e-8be1-884c4e116226': this.registerNode(arg)};
```

Comment:
@AlanFoster It's "random" - the WebDriver spec uses a similar UUID -https://www.w3.org/TR/webdriver/#elements - to indicate element references in command responses but states that the value should be a UUID that uniquely identifies the element.  capybara-webkit doesn't currently use a UUID value to uniquely identify elements so I went with a different key.